.github/workflows/main.yml

```diff
@@ -66,6 +66,10 @@ jobs:
       - name: Run tests
         run: |
           make test
+      - name: Typechecking
+        if: ${{ startsWith(runner.os, 'macOS') }}
```

Comment:
Unlike build + test, which needs the execution of the code, type-checking just reads the code and checks the coherency. It should not be OS-dependent, and I thought running once in whichever environment should suffice.

It could be possible that type-checking results might depend on OS if we use a system library, but I don't think any dependencies are system-sensitive.